packages/block-library/src/post-content/edit.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityProp, useEntitySaving } from '@wordpress/core-data';
+import { useMemo, useCallback } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { serializeBlocks } from '@wordpress/editor';
+
+export default function PostContentEdit() {
+	const [ content, setContent ] = useEntityProp( 'postType', 'post', 'content' );
+	const initialBlocks = useMemo( () => {
+		const parsedContent = parse( content );
+		return parsedContent.length ? parsedContent : undefined;
+	}, [] );
+	const [ blocks = initialBlocks, setBlocks ] = useEntityProp(
+		'postType',
+		'post',
+		'blocks'
+	);
+	const [ isDirty, isSaving, save ] = useEntitySaving(
+		'postType',
+		'post',
+		'content'
+	);
+	return (
+		<>
+			<Button
+				isPrimary
+				className="wp-block-custom-entity__save-button"
+				disabled={ ! isDirty || ! content }
+				isBusy={ isSaving }
+				onClick={ useCallback( () => {
+					setContent( content( { blocks } ) );
+					save();
+				}, [ content, blocks ] ) }
+			>
+				{ __( 'Update' ) }
+			</Button>
+			<InnerBlocks
+				value={ blocks }
+				onChange={ setBlocks }
+				onInput={ useCallback( () => {
+					setContent( ( { blocks: blocksForSerialization = [] } ) =>
+						serializeBlocks( blocksForSerialization )
+					);
```

Comment:
Not a concern with this code, but I'm curious how this would translate to an archive context. It absolutely makes sense to start with the single post context for this exploration, and in that context combining editing of the template blocks and post blocks should work fine. But once we get to a place where there are multiple posts in one page (and the list would change over time), the UI might fill up quite a bit. This could be reduced by some kind of "zoom-in" mode where post content is mostly hidden and can be expanded on click and/or with a link to "edit this specific post" (I believe this was considered before somewhere). Another consideration _might_ be performance - having blocks from let's say 10+ posts (plus the template blocks) in one view could be a lot, especially when assuming most people would only use the archive view for editing the template, but not individual posts.

Reply:
Yeah, if the editor is not operating on a single post, the post block can easily infer that from `useEntityId` returning undefined, and bail out of loading anything, showing a placeholder instead.